app/graphql/types/models/user_type.rb

```diff
@@ -0,0 +1,138 @@
+    field :email_general_feature, Boolean, null: false
+    field :email_general_commercial, Boolean, null: false
+    field :email_general_question, Boolean, null: false
+    field :email_html, Boolean, null: false
+    field :email_locations_admin, Boolean, null: true
+    field :email_names_admin, Boolean, null: true
+
+    # TODO: uncomment each association as the models are added
+
+    # belongs to
+    # field :image, Types::Models::ImageType, null: true
+    # field :license, Types::Models::LicenseType, null: false
+    # field :location, Types::Models::LocationType, null: true
+
+    # has many
+    # field :api_keys, [Types::Models::ApiKeyType], null: true
+    # field :comments, [Types::Models::CommentType], null: true
+    # field :donations, [Types::Models::DonationType], null: true
+    # field :external_links, [Types::Models::ExternalLinkType], null: true
+    # field :images, [Types::Models::ImageType], null: true
+    # field :interests, [Types::Models::InterestType], null: true
+    # field :locations, [Types::Models::LocationType], null: true
+    # field :location_descriptions, [Types::Models::LocationDescriptionType], 
+    #                               null: true
+    # field :names, [Types::Models::NameType], null: true
+    # field :name_descriptions, [Types::Models::NameDescriptionType], null: true
+    # field :namings, [Types::Models::NamingType], null: true
+    # field :notifications, [Types::Models::NotificationType], null: true
+    # field :observations, [Types::Models::ObservationType], null: true
+    # field :projects_created, [Types::Models::ProjectType], null: true
+    # field :publications, [Types::Models::PublicationType], null: true
+    # # field :queued_emails, [Types::QueuedEmailType], null: true
+    # field :sequences, [Types::Models::SequenceType], null: true
+    # field :species_lists, [Types::Models::SpeciesListType], null: true
+    # field :herbarium_records, [Types::Models::HerbariumRecordType], null: true
+    # field :votes, [Types::Models::VoteType], null: true
+    # field :reviewed_images, [Types::Models::ImageType], null: true
+    # field :reviewed_name_descriptions, [Types::Models::NameDescriptionType], 
+    #                                    null: true
+    # # field :to_emails, [Types::Models::QueuedEmailType], null: true
+
+    # has and belongs to many
+    # field :user_groups, [Types::Models::UserGroupType], null: true
+    # field :authored_names, [Types::Models::NameDescriptionType], null: true
+    # field :edited_names, [Types::Models::NameDescriptionType], null: true
+    # field :authored_locations, [Types::Models::LocationDescriptionType], 
+    #                            null: true
+    # field :edited_locations, [Types::Models::LocationDescriptionType], 
+    #                          null: true
+    # field :curated_herbaria, [Types::Models::HerbariumType], null: true
+
+    # custom fields
+    # field :img_src_thumb, String, null: true
```

Comment:
What are all of these img_xxx fields going to be for?  They might make sense in the image model, but not here in the user... or am I missing something?

Reply:
They all just call the Image.url method. I guess those URLs are created programatically in Rails - and they could be created programatically on the front end from the Image id, but in case the Rails method changes in the future, i thought it made sense to have this on the Rails side. (Hmmm. Maybe it should be one field returning an array...)

View wise, they deliver a `src` for every size we have (using HTML `srcset` allows a modern browser to download the smallest image necessary for the space occupied by the img at current screen size.) For a "matrix box" carousel with a fullscreen mode, if that's desirable, for example, you'd encode all those sizes in the `srcset`

Reply:
Oh - the reason they're here is just so we don't have to query the (user profile) image from the image table. Calling Image.url on the User.image_id gives the urls without hitting the image table.

But i think i understand what you mean, they could still all be in the Image model as methods!